engine-precompiles/src/modexp.rs

```diff
@@ -1,11 +1,29 @@
 use crate::prelude::types::{Address, EthGas};
 use crate::prelude::{PhantomData, Vec, U256};
-use crate::{
-    utils, Berlin, Byzantium, EvmPrecompileResult, HardFork, Precompile, PrecompileOutput,
-};
+use crate::{Berlin, Byzantium, EvmPrecompileResult, HardFork, Precompile, PrecompileOutput};
+use core::{cmp::min, mem::size_of};
 use evm::{Context, ExitError};
 use num::{BigUint, Integer};
 
+// this macro was originaly from https://github.com/bluealloy/revm/blob/8c22748b484ea2eb19d5200cc81ba7a4c3d7f8c7/crates/precompile/src/modexp.rs#L49
+macro_rules! read_u64_with_overflow {
+    ($input:expr,$from:expr,$to:expr, $overflow_limit:expr) => {{
+        const SPLIT: usize = 32 - size_of::<u64>();
+        let len = $input.len();
+        let from_zero = min($from, len);
+        let from = min(from_zero + SPLIT, len);
+        let to = min($to, len);
+        let overflow_bytes = &$input[from_zero..from];
+
+        let mut len_bytes = [0u8; size_of::<u64>()];
+        len_bytes[..to - from].copy_from_slice(&$input[from..to]);
+        #[allow(clippy::as_conversions)]
+        let out = u64::from_be_bytes(len_bytes) as usize;
```

Comment:
This conversion is not safe in Wasm because `usize` there is 32 bit. I also don't think we should be taking macros from other codebases.

Reply:
Thanks for pointing out to the actual issue. I will try to reproduce it with a fix shortly.